core/src/main/java/de/danoeh/antennapod/core/storage/DBTasks.java

```diff
@@ -334,11 +334,16 @@ private static Feed searchFeedByIdentifyingValueOrID(PodDBAdapter adapter,
     }
 
     /**
-     * Get a FeedItem by its identifying value.
+     * Get a FeedItem by its identifying value or download_url.
      */
-    private static FeedItem searchFeedItemByIdentifyingValue(Feed feed, String identifier) {
+    private static FeedItem searchFeedItemByIdentifyingValue(Feed feed, FeedItem searchItem) {
         for (FeedItem item : feed.getItems()) {
-            if (TextUtils.equals(item.getIdentifyingValue(), identifier)) {
+            if (TextUtils.equals(item.getIdentifyingValue(), searchItem.getIdentifyingValue())) {
+                return item;
+            }
+            if (!TextUtils.isEmpty(item.getLink())
```

Comment:
I would only do the fallback checks if all other items have a non-matching identifying value. Wouldn't using the media download url be more reliable than the website link?

Another thing we could do would be to test if title **and** date match. Then we can catch publishers who change both guid and media file url. I noticed some publishers doing that recently.

Reply:
Using the download url now